python/desc/skycatalogs/objects/base_object.py

```diff
@@ -217,8 +218,14 @@ def get_sed(self, component=None, resolution=None):
                                  self.get_native_attribute('sed_filepath'))
 
             return sky_cat.observed_sed_factory.create_pointsource(rel_path), mag_norm
+        elif self._object_type == 'sn':
+            # Make an SNObject and get SED
+            params = self.get_native_attribute('salt2_params')
+            sn = SNObject(params=params)
```

Comment:
Since supernovae have non-zero fluxes over limited time intervals, i.e., ~70 days or so, I had been thinking that we should check when the SN fluxes are non-zero and return `None` for the SED outside of that time period.  Looking through the [documentation](https://sncosmo.readthedocs.io/en/stable/api/sncosmo.Model.html#sncosmo.Model), I see that there are `.mintime()` and `.maxtime()` methods that return the bounds of when the model is defined.  So we should add something along the lines of
```
if mjd < sn.mintime() or mjd > sn.maxtime():
    return None, 0.0
```
here.
However, I'm concerned about how an `sed` that equals `None` is used later in functions like `.get_flux(...)`.  There seems to be an underlying assumption that `sed is None` would only occur for galaxies and there is some special handling that I don't understand in those cases.

Reply:
There are cases where the disk (I think) component of a galaxy was negligible so no SED was computed but there was something for bulge.  I think that's where the special handling for galaxies came in.  There was maybe still the implicit assumption that *some* component had a SED, so downstream code has not been exercised in the case where there is no SED at all.

Reply:
I think it's a matter of checking for a return of "None" from get_total_observer_sed and handling it appropriately. It's called in various routines for calculating fluxes.  They could check for "None" and return zero flux (or fluxes) in that case. 
Also get_total_observer_sed itself needs to check for sed == none before trying to multiply the sed by mu.   I'll put these changes in as soon as NERSC is back

Reply:
There are places where sed == None is taken to mean "not yet computed" so some adjustments need to be made, in particular any place where get_total_observer_sed is called.     The caller has to look for a return of None and handle it properly.  For flux routines I assume they should just return 0 fluxes.  Also within get_total_observer_sed itself there needs to be a check for sed == None before attempting to multiply by mu.